dan_layer/consensus/src/hotstuff/on_propose.rs

```diff
@@ -144,11 +171,14 @@ where
         const TARGET_BLOCK_SIZE: usize = 1000;
         let mut remaining_txs = TARGET_BLOCK_SIZE;
 
-        let commit_txs = PrecommitTransactionPool::move_many_to_committed(tx, remaining_txs)?;
+        // TODO: Decide if we should lock the txs to the next block somehow
```

Comment:
This should be fine (tm) since the replica will receive this block via loopback and it should in almost all cases be locked anyway